### PR TITLE
Add Symfony environment to commands executed in ScriptHandler

### PR DESCRIPTION
**Pull request by javihgil**
When using composer install command with --no-dev flag, ScriptHandler executes commands without specify environment. 

So, if you have a development dependency in require-dev and this development bundle is loaded in AppKernel only for dev and test environment, when you execute install with --no-dev option you will get an error.

For example, I want to move "sensio/generator-bundle" from "require" to "require-dev" in composer.json, because of I don't want to this code generators be able in production servers. 

``` json
"require-dev": {
    "sensio/generator-bundle": "~3.0"
},
```

In AppKernel this bundle is loaded only for dev and test Symfony environments. 

``` php
if (in_array($this->getEnvironment(), ['dev', 'test'])) {
    $bundles[] = new Symfony\Bundle\DebugBundle\DebugBundle();
    $bundles[] = new Symfony\Bundle\WebProfilerBundle\WebProfilerBundle();
    $bundles[] = new Sensio\Bundle\DistributionBundle\SensioDistributionBundle();
    $bundles[] = new Sensio\Bundle\GeneratorBundle\SensioGeneratorBundle();
}
```

If I do this, when I execute composer install --no-dev command I get a ScriptHandler::clearCache error because it makes an "app/console cache:clear --no-warmup" without specify environment, so "dev".

```
> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache
Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the post-install-cmd event terminated with an exception

  [RuntimeException]                                                         
  An error occurred when executing the "'cache:clear --no-warmup'" command
```

**Proposal**
I propose a new option for script handler witch provides this feature in different ways:

You can specify env in composer.json extra:

``` json
"extra": {
    "symfony-env": "prod"
},
```

You can do it with an environment variable:

``` bash
SYMFONY_ENV=prod composer install --no-dev
```

### CODE REVIEW

File: Composer/ScriptHandler.php [hunk 1]
```diff
@@ -30,6 +30,7 @@ class ScriptHandler
     protected static $options = array(
         'symfony-app-dir' => 'app',
         'symfony-web-dir' => 'web',
+        'symfony-env' => false,
```
Comment (HeahDude) [inline Composer/ScriptHandler.php]: could you set `'dev'` by default, it makes more sense than `false`.

File: Composer/ScriptHandler.php [hunk 2]
```diff
@@ -301,6 +302,11 @@ protected static function executeCommand(Event $event, $consoleDir, $cmd, $timeo
             $console .= ' --ansi';
         }
 
+        $options = static::getOptions($event);
+        if ($options['symfony-env']) {
```
Comment (HeahDude) [inline Composer/ScriptHandler.php]: I think you can inline here we don't further need the options:
```php
if ('dev' !== $env = static::getOptions($event)['symfony-env']) {
   $cmd .= '--env='.$env;
}
```

File: Composer/ScriptHandler.php [hunk 3]
```diff
@@ -394,6 +400,8 @@ protected static function getOptions(Event $event)
 
         $options['symfony-assets-install'] = getenv('SYMFONY_ASSETS_INSTALL') ?: $options['symfony-assets-install'];
 
+        $options['symfony-env'] = getenv('SYMFONY_ENV') ?: $options['symfony-env'];
```
Comment (HeahDude) [inline Composer/ScriptHandler.php]: can you pass this just before `symfony-assets-install` without blank line between them?

**Comment by HeahDude** [comment]
If we intend to support all environments installation, I'm 👍 for this

**Comment by javihgil** [comment]
Thanks @HeahDude for your suggestions, I made the changes

File: Composer/ScriptHandler.php [hunk 4]
```diff
@@ -428,8 +434,8 @@ protected static function getPhpArguments()
     /**
      * Returns a relative path to the directory that contains the `console` command.
      *
-     * @param Event $event      The command event.
-     * @param string       $actionName The name of the action
+     * @param Event  $event      The command event.
+     * @param string $actionName The name of the action
```
Comment (HeahDude) [inline Composer/ScriptHandler.php]: These two lines of cs fix are now merged in master, can you please revert them and rebase? Thanks :)

**Comment by HeahDude** [comment]
LGTM 👍 

What do you think @fabpot?

File: Composer/ScriptHandler.php (hunk 4 above)
Comment (javihgil) [inline Composer/ScriptHandler.php]: I just updated upstream and changes are not merged yet:

https://github.com/sensiolabs/SensioDistributionBundle/blob/master/Composer/ScriptHandler.php

Comment (HeahDude) [inline Composer/ScriptHandler.php]: Yeah I saw, but it'll be soon, thanks for being thorough :)

**Comment by lordt** [comment]
Hi @javihgil , can you apply change suggested by bot?

**Comment by fabpot** [comment]
Closing this pull request as this bundle is now deprecated. Symfony Flex replaces this bundle.

**Pull request closed without merging**